lib/matplotlib/units.py

```diff
@@ -134,73 +134,33 @@ def is_numlike(x):
 
 
 class Registry(dict):
-    """
-    A register that maps types to conversion interfaces.
-    """
-    def __init__(self):
-        dict.__init__(self)
-        self._cached = {}
+    """Register types with conversion interface."""
 
     def get_converter(self, x):
-        """
-        Get the converter for data that has the same type as *x*. If no
-        converters are registered for *x*, returns ``None``.
-        """
-
-        if not len(self):
-            return None  # nothing registered
-        # DISABLED idx = id(x)
-        # DISABLED cached = self._cached.get(idx)
-        # DISABLED if cached is not None: return cached
-
-        converter = None
-        classx = getattr(x, '__class__', None)
-
-        if classx is not None:
-            converter = self.get(classx)
-
-        if converter is None and hasattr(x, "values"):
-            # this unpacks pandas series or dataframes...
-            x = x.values
-
-        # If x is an array, look inside the array for data with units
+        """Get the converter interface instance for *x*, or None."""
+        if hasattr(x, "values"):
```

Comment:
@ImportanceOfBeingErnest had some concerns about this way to check for pandas: 
https://github.com/matplotlib/matplotlib/pull/11664#discussion_r202550193

Reply:
One could do something like `type(x).__module__.startswith("pandas.") and hasattr(x, "values")` but that should be a separate PR; this PR doesn't change the way pandas testing is done.